RDCP: Sort by system modified date in browse by collection view

mcritchlow opened the pull request:
Fixes #644 

#### Local Checklist
- [x] Tests written and passing locally?
- [x] Code style checked?
- [x] QA-ed locally?
- [x] Rebased with `master` branch?
- [ ] Configuration updated (if needed)?
- [ ] Documentation updated (if needed)?

#### What does this PR do?
- Sorts rdcp browse by collection view by `system_modified` date descending, with `title` ascending as a secondary sort parameter
- Adds tests demonstrating this behavior for RDCP, but **not** for other units (DLDP)
- Adds `timecop` gem to easily support time-sensitive tests, which were relevant for the behavior in this PR
- Fixed a leaky test unit object that showed up in a [failed CI run](https://circleci.com/gh/ucsdlib/damspas/1511). This is not completely related, but seems problematic and worth fixing (I worry we have other examples of this..)

##### Why are we doing this? Any context of related work?
References #644 - for RDCP

@ucsdlib/developers - please review

lsitu commented on spec/features/collections_search.rb:
```diff
@@ -0,0 +1,57 @@
+      old_collection_sort_order = page.body.index('A Old Collection')
+      new_collection_sort_order = page.body.index('The New Collection')
+      expect(old_collection_sort_order).to be > new_collection_sort_order
+    end
```
Could we add a test for user selected sorting rules like` sort by collection title` to see whether it works? It seems like there could be a gap but I may be wrong. See my other comment above.

mcritchlow replied:
The default sort I think is tested in the DLDP browse test I wrote below the RDCP one, which basically demonstrates the normal alphabetical sort. Can you take a look at that again and let me know if you think there's still a gap that needs testing? It's certainly possible :)

mcritchlow replied:
@lsitu - Oh, I apologize. You're saying if a user selects different sorting rules after the page has initially loaded. That's a really great point, and possibly a regression. The current code might always force this sorting for RDCP. I understand what you're saying now, thanks for pointing this out. I'll think about how to adjust for this.

mcritchlow commented on app/controllers/catalog_controller.rb:
```diff
@@ -307,6 +307,10 @@ def collection_search
+
+    # RDCP prefers their collections sorted by most recently created first
+    # Therefore, we're sorting by modified date rather than title
+    params[:sort] = 'system_modified_dtsi desc, title_ssi asc' if params[:id].eql? 'rdcp'
```
My thinking was that this would be last in the chain of sorting checks. If I put it above, then it could potentially get overwritten later. But I may be misunderstanding you, can you help clarify with an example maybe?

lsitu replied:
@mcritchlow I think it's just related to your last comment in https://github.com/ucsdlib/damspas/pull/663#discussion_r295917873. The default `sort by title` will be set with the `unless params[:sort]` in https://github.com/ucsdlib/damspas/pull/663/files#diff-956e46b33fb5307990c4e1a4b5fd86ccR309 only when there are no sort order rules are selected.

mcritchlow commented:
@lsitu - Thank you again for catching the user-selected sorting regression! I've changed the code in catalog controller to only apply those defaults if no sorting option has been selected by the user. And I added a test to confirm that.

Let me know if you think this works.

hweng commented:
:+1:
